lib/cli/index.js

```diff
@@ -7,6 +7,10 @@ var logger        = require('heimdalljs-logger')('ember-cli:cli/index');
 var merge         = require('ember-cli-lodash-subset').merge;
 var path          = require('path');
 
+// work around misbehaving libraries, so we can correctly cleanup before
+// actually exiting.
+require('capture-exit').captureExit();
```

Comment:
@nathanhammond what are you not concerned about?  @Turbo87's suggestion to move this before other requires?

His point seems valid; we want to `captureExit` as soon as possible; this is unrelated to when an interruption occurs because we want to get head not only of interruptions but also people who pluck off `process.exit`